app/code/Magento/ProductAlert/Controller/Unsubscribe/Stock.php

```diff
@@ -60,6 +60,10 @@ public function execute()
                     $this->_objectManager->get(\Magento\Store\Model\StoreManagerInterface::class)
                         ->getStore()
                         ->getWebsiteId()
+                )->setStoreId(
+                    $this->_objectManager->get(\Magento\Store\Model\StoreManagerInterface::class)
```

Comment:
Please move this dependency to constructor.

Reply:
yes sure , i will add it in construct according to backward compatibility.
Also will replace `\Magento\ProductAlert\Model\Stock` to `\Magento\ProductAlert\Model\StockFactory` and failed test cases.

Thank you.